test/test_torch.py

```diff
@@ -9397,6 +9397,40 @@ def test_split_with_sizes_copy_out(self):
             for expect, t in zip(expects, out):
                 self.assertTrue(expect.eq(t).all().item())
 
+    def test_chunk_cat(self):
+        def pad_tensor(tensor: torch.Tensor, dim: int, num_chunks: int) -> torch.Tensor:
+            input_shape = list(tensor.size())
+            size_along_dim = input_shape[dim]
+            padded_size_along_dim = (size_along_dim + num_chunks - 1) // num_chunks * num_chunks
+            padded_size = input_shape[:dim] + [padded_size_along_dim,] + input_shape[dim + 1 :]
+            padded_tensor = torch.zeros(padded_size, device=tensor.device, dtype=tensor.dtype)
+            padded_tensor.narrow(dim=dim, start=0, length=size_along_dim).copy_(tensor)
+            return padded_tensor
+
+        def chunk_cat(tensors, dim, num_chunks):
```

Comment:
If you want to check this reference implementation, you can provide it as reference implementation to the opinfo and this test will run automatically.

I do suspect that we already check that decomp matches kernel impl via OpInfo so this might not be needed at all if the decomposition you have is already the reference implementation.